### ISSUE TITLE
[BUG/Help] <单卡运行请求返回乱码>

### ISSUE THREAD

**Issue by scauwjh** [THUDM/ChatGLM2-6B#224]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

请求 “你好”，给我返回了这些东西，且计算时间比较长。16G显存，跑的是chatglm2-6b模型
02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02\x02

### Steps To Reproduce

直接运行就报错了

### Environment

```markdown
- OS: centos7
- Python: python3.10
- Transformers: 
- PyTorch: 2.0+cu117
- CUDA Support (`python -c "import torch; print(torch.cuda.is_available())"`) :
```

**Comment by lilongxian** [comment]
再紧接着问一句，回复就好了

**Comment by lilongxian** [comment]
1. 建议环境为：
  CUDA 117。 Python3.7-3.8  pytorch 1.13.x /2.0
transformers 4.27.1---4.29.2

2. 另外tokenization py脚本中的解码函数decode()：
<html>
<body>


def decode(self, t: List[int]) -> str:
  return self.sp_model.decode(t) 
建议修改为：
    def decode(self, t: List[int]) -> str:
        items = list(self.special_tokens.values()) + [-100]
        for i in items:
            t = list(filter((i).__ne__, t))
        return self.sp_model.decode(t)

       过滤掉 {'[MASK]': 64789, '[gMASK]': 64790, '[sMASK]': 64791, 'sop': 64792, 'eop': 64793}
        中的特殊字符以及-100（表示pad）看看。

**Comment by scauwjh** [comment]
> 1. 建议环境为：
>    CUDA 117。 Python3.7-3.8  pytorch 1.13.x /2.0
>    transformers 4.27.1---4.29.2
> 2. 另外tokenization py脚本中的解码函数decode()：
> 
> def decode(self, t: List[int]) -> str:   return self.sp_model.decode(t) 建议修改为： def decode(self, t: List[int]) -> str: items = list(self.special_tokens.values()) + [-100] for i in items: t = list(filter((i).**ne**, t)) return self.sp_model.decode(t)
> 
> ```
>    过滤掉 {'[MASK]': 64789, '[gMASK]': 64790, '[sMASK]': 64791, 'sop': 64792, 'eop': 64793}
>     中的特殊字符以及-100（表示pad）看看。
> ```

改了没效果

**Comment by scauwjh** [comment]
估计是因为我模型不是从huggingface上下载下来的原因。我是从清华的云盘下的模型文件=_=||

**Comment by yl3118436** [comment]
> 估计是因为我模型不是从huggingface上下载下来的原因。我是从清华的云盘下的模型文件=_=||

我也是和你同样的问题，请问解决了吗？

**Comment by scauwjh** [comment]
> > 估计是因为我模型不是从huggingface上下载下来的原因。我是从清华的云盘下的模型文件=_=||
> 
> 我也是和你同样的问题，请问解决了吗？

删掉直接huggingface下载，或者直接脚本自己缓存下载
